File: scripts/deploy_whitepaper.py
import os
import re
import shutil


PAGE_BREAK_DIV_PATTERN = re.compile(
    r'^[ \t]*<div\s+style=["\']page-break-after:\s*always;?["\']\s*>\s*</div>\s*\r?\n?',
    re.IGNORECASE | re.MULTILINE
)


def sanitize_markdown_for_web_gui(content):
    return PAGE_BREAK_DIV_PATTERN.sub("", content)


def copy_assets(source_dir, target_dir, allowed_extensions):
    if not os.path.isdir(source_dir):
        print(f"Source directory not found: {source_dir}")
        return 0

    if os.path.exists(target_dir):
        shutil.rmtree(target_dir)
    os.makedirs(target_dir, exist_ok=True)

    count = 0
    for name in os.listdir(source_dir):
        source_path = os.path.join(source_dir, name)
        if not os.path.isfile(source_path):
            continue
        ext = os.path.splitext(name)[1].lower()
        if ext not in allowed_extensions:
            continue
        target_path = os.path.join(target_dir, name)
        if ext == ".md":
            with open(source_path, "r", encoding="utf-8") as source_file:
                source_content = source_file.read()
            with open(target_path, "w", encoding="utf-8") as target_file:
                target_file.write(sanitize_markdown_for_web_gui(source_content))
        else:
            shutil.copy2(source_path, target_path)
        count += 1

    print(f"Deployed {count} assets to {target_dir}")
    return count


def deploy_whitepaper():
    project_root = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

    whitepaper_src = os.path.join(project_root, "docs", "whitepaper")
    whitepaper_dest = os.path.join(project_root, "app", "public", "whitepaper")
    comic1_src = os.path.join(project_root, "docs", "comic1")
    comic1_dest = os.path.join(project_root, "app", "public", "comic1")
    comic2_src = os.path.join(project_root, "docs", "comic2")
    comic2_dest = os.path.join(project_root, "app", "public", "comic2")
    comic3_src = os.path.join(project_root, "docs", "comic3")
    comic3_dest = os.path.join(project_root, "app", "public", "comic3")

    total = 0
    total += copy_assets(whitepaper_src, whitepaper_dest, {".md", ".png", ".jpg", ".jpeg", ".svg", ".pdf", ".mp4"})
    total += copy_assets(comic1_src, comic1_dest, {".png", ".jpg", ".jpeg"})
    total += copy_assets(comic2_src, comic2_dest, {".png", ".jpg", ".jpeg"})
    total += copy_assets(comic3_src, comic3_dest, {".png", ".jpg", ".jpeg"})

    print(f"Deployment complete. {total} whitepaper assets deployed.")


if __name__ == "__main__":
    deploy_whitepaper()

File: scripts/deploy_whitepaper.ts
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const ROOT_DIR = path.resolve(__dirname, '..');

// Configuration
const MAPPINGS = [
    {
        source: path.join(ROOT_DIR, 'docs', 'whitepaper'),
        target: path.join(ROOT_DIR, 'app', 'public', 'whitepaper'),
        extensions: ['.md', '.png', '.jpg', '.jpeg', '.svg', '.pdf']
    },
    {
        source: path.join(ROOT_DIR, 'docs', 'comic1'),
        target: path.join(ROOT_DIR, 'app', 'public', 'comic1'),
        extensions: ['.png', '.jpg', '.jpeg']
    },
    {
        source: path.join(ROOT_DIR, 'docs', 'comic2'),
        target: path.join(ROOT_DIR, 'app', 'public', 'comic2'),
        extensions: ['.png', '.jpg', '.jpeg']
    }
];

const PAGE_BREAK_DIV_REGEX = /^[ \t]*<div\s+style=["']page-break-after:\s*always;?["']\s*>\s*<\/div>\s*\r?\n?/gim;

const sanitizeMarkdownForWebGui = (content: string) => (
    content.replace(PAGE_BREAK_DIV_REGEX, '')
);

function copyAssets(sourceDir: string, targetDir: string, allowedExtensions: string[]) {
    if (!fs.existsSync(sourceDir)) {
        console.warn(`Source directory not found: ${sourceDir}`);
        return 0;
    }

    // Create target directory (clearing it first might be safer, but simpler: just ensure exists)
    // Python script did: if exists, rmtree; then makedirs. Let's match that behavior.
    if (fs.existsSync(targetDir)) {
        fs.rmSync(targetDir, { recursive: true, force: true });
    }
    fs.mkdirSync(targetDir, { recursive: true });

    const items = fs.readdirSync(sourceDir);
    let count = 0;

    for (const name of items) {
        const sourcePath = path.join(sourceDir, name);
        const stat = fs.statSync(sourcePath);

        if (stat.isFile()) {
            const ext = path.extname(name).toLowerCase();
            if (allowedExtensions.includes(ext)) {
                const targetPath = path.join(targetDir, name);
                if (ext === '.md') {
                    const sourceContent = fs.readFileSync(sourcePath, 'utf8');
                    fs.writeFileSync(targetPath, sanitizeMarkdownForWebGui(sourceContent), 'utf8');
                } else {
                    fs.copyFileSync(sourcePath, targetPath);
                }
                count++;
            }
        }
    }

    console.log(`Deployed ${count} assets to ${targetDir}`);
    return count;
}

function deployWhitepaper() {
    console.log('Starting whitepaper deployment...');
    let total = 0;

    for (const mapping of MAPPINGS) {
        total += copyAssets(mapping.source, mapping.target, mapping.extensions);
    }

    console.log(`Deployment complete. ${total} whitepaper assets deployed.`);
}

deployWhitepaper();
